subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -51,14 +51,10 @@ public decimal GetYearlyCost(Subscription subscription)
 
     public IEnumerable<Subscription> GetSubscriptionListByMonth(IEnumerable<Subscription> subscriptions, DateTime monthDate)
     {
-        bool paidThisMonth(Subscription s) => s.LastPayment.Month == monthDate.Month && s.LastPayment.Year == monthDate.Year;
-        bool paymentStartsInTheFuture(Subscription s) => s.LastPayment > monthDate;
-
-        var subscriptionsByMonth = subscriptions
-                    .Where(s => !paidThisMonth(s) && !paymentStartsInTheFuture(s))
-                    .ToList();
-
-        return subscriptionsByMonth;
+        return subscriptions
+                    .Where(subscription => PaidInPast(subscription, monthDate))
+                    .SelectMany(subscription => GetIterationsForMonth(subscription, monthDate))
+                    .OrderBy(subscription => subscription.LastPayment);
```

Comment:
Head in to the Monthly.razor and remove the unneeded order by there.  Linq is beautiful